maturity-level-two/docs/api-security.md

```diff
@@ -0,0 +1,127 @@
+
+The advantage over the previously explained options is that this can give you a certainty over a user, and if the certificates are securely passed from the one party to the other, you have a very secure system.
+
+If you want to implement this in .NET Core, you can speed up things by using the Arcus NuGet package as described [here](https://webAPI.arcus-azure.net/features/security/auth/certificate).
+
+### OAuth2
+OAuth2 has various implementations. From IdentityServer to manage the users/secrets by yourself or Facebook, Google or Microsoft (with Azure AD) where the whole user management is done by the OAuth2 provider itself.
+
+This setup might require you the most effort but is also the most secure (if you choose only one).
+
+The principle is that the user will authenticate to the OAuth2 Authentication server. The server will validate if your credentials are valid credentials for this certain application. If that is the case an `Authorization`-header with a JWT token can be added to the request. The API can then validate if:
+- The user is authentication with the right Authorization Server
+- What the email & basic information of the user is.
+- The roles the user has (based on claims)
+
+And do the appropriate action custom made for that user.
+
+Bearer tokens can only be generated by the authentication server so there is no security risk there.
+Usually a token is only valid for x amount of time and will expire. Every time a user wants to start a new session he needs to generate a new bearer token via the Authentication server.
+
+Usually within the Authentication Server you can easily add and remove users for your application.
+
+## Azure Specifics
+Azure can give you some out of the box features to improve the security of your system. In the list below you can see some examples how this can help.
+
+### API Management
+- Https is by default there on API management
+- [IP-Filtering can be enabled via policy](https://docs.microsoft.com/en-us/azure/API-management/API-management-access-restriction-policies#RestrictCallerIPs)
+- [Client certificate validation can be done via policies](https://docs.microsoft.com/en-us/azure/API-management/API-management-howto-mutual-certificates-for-clients)
+- [API-Keys can be added via a policy](https://docs.microsoft.com/en-us/azure/API-management/API-management-access-restriction-policies#CheckHTTPHeader)
+- [Basic authentication validation can be added via a policy](https://docs.microsoft.com/en-us/azure/API-management/API-management-authentication-policies#Basic)
+- [API management also has a system of users and subscribers based on API-Keys which allows you to rotate keys, and handle creation and disabling of users by a user interface/powershell scripts.](https://docs.microsoft.com/en-us/azure/API-management/API-management-subscriptions)
```

Comment:
I see, I would merge this with the Check header suggestion and set the context around those a bit when to use what

Reply:
Not sure what you mean here?

Reply:
I don't remember, it's been a while so I'm resolving it 🤷‍♂️

Reply:
Actually it was about the suggestion to use https://docs.microsoft.com/en-us/azure/API-management/API-management-access-restriction-policies#CheckHTTPHeader where this is done manually. Now, frankly, I do not know why you would want to build your own via https://docs.microsoft.com/en-us/azure/API-management/API-management-access-restriction-policies#CheckHTTPHeader rather than using the built-in approach

Reply:
You are two times using the same url. not sure what you mean? Are you talking about compared to the User/subscription model of API-M?